### PR TITLE
PR for M3 r1.1

### PR DESCRIPTION
**Pull request by Kevsy**
Fixed some link syntax

#### What type of PR is this?

Add one of the following kinds:
* documentation : fixed some link syntax in CHANGELOG

#### What this PR does / why we need it:

PR for r1.1 - rc

#### Which issue(s) this PR fixes:

Meets release scope per #63

#### Special notes for reviewers:
This is the proposed PR for the M3 release

### CODE REVIEW

**Comment by Kevsy** [comment]
Please could Release Management kindly review this PR to allow merge to main / release with tag r1.1

**Comment by hdamker** [comment]
Thanks @Kevsy ... I have created https://github.com/camaraproject/ReleaseManagement/issues/74 for the review which I will take next week.

I see that the CHANGELOG is already partly updated in advance, and also the version within in the YAML files are already set (should have happen in the release PR, but ok). 

Few points which you can already address here in the PR:
* the update of the README within the section Release Information (it has to be done in the Release PR as it is otherwise not part of the release tag)
* add in the Changelog the section for the application-profile API
* add the links to the .feature files in the API Readiness Checklist

Have a nice weekend!

**Comment by Kevsy** [comment]
Thanks @hdamker , changes to address your points now committed.

**Comment by hdamker** [comment]
@Kevsy please delete the tag r1.1 which you created four days ago. It points to the merge commit of #66 which is NOT the approved release candidate. The correct process is described step by step here: https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/API_Release_Guidelines.md#releasing-an-api-step-by-step.

**Comment by hdamker** [comment]
Please find my review comments on connectivity-insights.yaml:

There are some critical deviations from the API Design Guidelines:

-  /network-quality/insights:
   - I don't understand the use of the of path element `network-quality` ... it would need to be a resource, but it isn't one.
      - If you think that this API is about network quality insights, why not calling the API `network-quality-insights` and the endpoint just /check or similar (it must be a verb, see below)? 
      - if you are fine with the API name `connectivity-insights`, why not just use a verb here for this endpoint? `check-network-quality` might be an option, avoiding the second level
   - the operation is actually not creating a resource, **the description is mentioning an `insightId` but that is not returned**. If it would create a resource, there must be read and delete operations in the API.
   - it seems to be a "GET insights" using a POST operation due to the sensible information in the body, in which case the following guideline applies (cf. https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/documentation/API-design-guidelines.md#post-or-get-for-transferring-sensitive-or-complex-data):

> When the POST method is used, the resource in the path must be a verb (e.g. retrieve-location and not location) to differentiate from an actual resource creation.

- subscriptions endpoints don't follow the guidelines in https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/documentation/API-design-guidelines.md#resource-based-explicit-subscription
  - /network-quality/insights/subscriptions does not work
  - a proposal could be to call the separate API network-quality-insights-subscriptions, the endpoint would be then just /subscriptions

> To ensure consistency across Camara subprojects, it is necessary that explicit subscriptions are handled within separate API/s. It is mandatory to append the keyword "subscriptions" at the end of the API name. For e.g. device-roaming-subscriptions.yaml

Further less critical comments:

- scope name for the network quality insight `connectivity_insights:network_quality_insight:write` does not follow https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/documentation/API-design-guidelines.md#1161-scope-naming, with defines the structure as `api-name:[resource:]action`
  - the api-name is `connectivity-insights`, not `connectivity_insights`
  - there is no `resource` involved, see above
  - For POST operations with a verb, the `action` will be a verb, not a CRUD operation.

- info.description - Authorization and Authentication: replace lines 62+63 with the updated text from ICM r0.2.0-rc.2 (link to main replaced). This can be done here in the PR or as an issue after the r1.1.
- info.version: `0.4.0-rc.1` instead of `"0.4.0-rc.1"` (cf [API-design-guidelines.md#info-object](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#info-object))
- info.contact is optional and and may be added by API Providers documenting their APIs

- responses:
  - CreateSubscriptionBadRequest400: "OUT_OF_RANGE" missing
  - Generic400: not aligned with https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/artifacts/CAMARA_common.yaml, "OUT_OF_RANGE" missing
  - CreateSubscription403: not aligned with https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/artifacts/camara-cloudevents/event-subscription-template.yaml, e.g. "INVALID_TOKEN_CONTEXT" missing, typo in line 1143 "SUSBCRIPTION_MISMATCH"
  - Device related error responses as defined in https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/documentation/API-design-guidelines.md#62-error-responses---device-object seem to be missing, e.g. in 422.
  - ... please check all error responses for alignment

**Comment by hdamker** [comment]
@Kevsy @maheshc01 Please see my comments above - I've stopped after the review of the connectivity-insights.yaml. How do you want to proceed? One option could be to first finish the connectivity-insights with one endpoint, and postpone the subscription API to a later release.

**Comment by Kevsy** [comment]
@hdamker Many thanks for the thorough review, and fully noted. Release tag has been deleted. Regarding the YAML we will confirm following the COI call later today.

Below my preferences based on the helpful advice above,:

> if you are fine with the API name connectivity-insights, why not just use a verb here for this endpoint? check-network-quality might be an option, avoiding the second level
- my preference `/connectivity-insights/{version}/check-network-quality`

> the operation is actually not creating a resource, the description is mentioning an insightId but that is not returned. 

- I think we should not create a resource, rather the following point applies here:

> it seems to be a "GET insights" using a POST operation due to the sensible information in the body, in which case the following guideline applies
- ...hence we should refactor according to the guideline

So those are the critical points for connectivity-insights.yaml , and the others will be addressed too.

> One option could be to first finish the connectivity-insights with one endpoint, and postpone the subscription API to a later release.

@Mahesh, we can deicide this on the call later

**Comment by hdamker** [comment]
@Kevsy thanks for the response. Regarding application-profiles there are also some comments and questions, but less severe. Should I add them here in similar way as above or would you prefer to have them in an issue?

**Comment by Kevsy** [comment]
> Should I add them here in similar way as above or would you prefer to have them in an issue?

Thanks @hdamker - please add them here for now, and then we can break them all out in separate issues to make sure they are all covered. We can do this after the call today.

**Comment by hdamker** [comment]
Please find my review comments on application-profiles.yaml:

- L9 - info.description: the description is quite short and less concrete then the tag description in line 28/29. Might make sense to mention that there are future extensions planned. The mandatory text about Authorization and Authentication from ICM is missing
- L36 - scope naming: `connectivity_insights:policy:write` -> `application-profiles:write` (`resource` is optional according to https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#1161-scope-naming, otherwise it would be `application-profiles:application-profiles:write`)
- L48 - Why `"*/*"`? The API Design Guidelines mandate `application/json` (comment applies also to connectivity-insights.yaml)
- L50 - Consider to define a schema like `CreateApplicationProfile` which contains today only `NetworkQualityThresholds` but can be extended with further optional fields (see also below).
- L53 - response should be 201 (Created), not 200
- L58 - Why returning only the ApplicationProfileId and not the full representation of the resource object? (that is inconsistent with `updateApplicationProfile` where the complete resource object is returned)
- L65 - 404 is not an expected response, as the resource will be just created
- L67 - 409 - which kind of "CONFLICT" do you expect here? How would a test case look like to generate this error message ?
- L75 - Consider "PUT" vs "PATCH". 
  - "PUT" will replace the complete resource object. It would also require some more definition what happens if 
    - a) the `applicationProfileId` in the path does not yet exist (PUT would allow to create it, but I don't think that's what you want here, as the ids are generated on the server side) and
    - b) if the `applicationProfileId` in the path and the `applicationProfileId`in the requestBody do not match (that should be an error response)
   - both topics can be avoided by using "PATCH" and in the requestBody `NetworkQualityThresholds` (or some `UpdateApplicationProfile` structure). That would be consistent with `createProfile` and with the description: "Update the network quality thresholds for an application ...". In this case you need to define that the `NetworkQualityThresholds` are replaced as a whole.
- L78 - scope name: `connectivity_insights:application_profile:write` -> `application-profile:write` (cf. L36)
- L82 - "Update the network quality thresholds for an application ..." -> "Update the network quality thresholds of an  application profile ..." (the profile can be used by multiple applications). Also the semantic of the PUT or PATCH should be defined, see comment on L75
- L88 - typo
- L97 - this description does not fit to "PUT" as PUT updates the entire resource with data that is passed in the body payload.
- L99 - see comment on line 48
- L129 - scope name, see above
- L159 - see comment on L67
- L169 - scope name, see above
- L195 - see comment on L67
- L313 ApplicationProfile definition:
  - shouldn't the applicationProfileId be required?
  - would an (optional) name (freely chosen by the API user) make sense?
- L358ff - responses: consider to copy the used responses from https://github.com/camaraproject/Commonalities/blob/main/artifacts/CAMARA_common.yaml for consistency (but most responses do not apply for this API)
  - L359 in Generic400 ... "OUT_OF_RANGE" is missing from CAMARA_common.yaml

**Comment by hdamker** [comment]
On the CHANGELOG.md:

There wasn't any previous release (at least not within CAMARA and in this CHANGELOG), so both APIs are new and have there first release here. If you want to compare to some previous version, please mention it (and where it can be found).

* L21/22: template proposed here `API-name1 vx.y.z` (with version number)
* L31: suggestion: `** 0.4.0-rc.1 is the first release-candidate version for the Connectivity-Insights 0.4.0**`
* L34: headline "Main changes" is not needed here, line 35 can follow directly after line 31. 
* L35: should be mentioned that his is the first release of the API

Same for application-profiles.

**Comment by Kevsy** [comment]
Thanks @hdamker - issues already created for the `connectivity-insights.yaml` changes noted above and PR incoming for approval by COI team. Once approved we will re-issue the 0.4.0-rc.1 PR for Release Management approval.

application-profiles issues/PR will follow.

**Comment by Kevsy** [comment]
New commit made to connectivity-insights.yaml per review feedback, fixes:
* camaraproject/ConnectivityInsights/issues/68
* camaraproject/ConnectivityInsights/issues/69
* camaraproject/ConnectivityInsights/issues/70
* camaraproject/ConnectivityInsights/issues/71
* camaraproject/ConnectivityInsights/issues/72
* camaraproject/ConnectivityInsights/issues/73

File: code/API_definitions/application-profiles.yaml
Comment (hdamker) [inline code/API_definitions/application-profiles.yaml:145]: Typo?

The ref goes still to 409, and 405 is not a valid response code for a defined method. See also https://github.com/camaraproject/Commonalities/issues/285

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml:145]: i missed updating the path to point to 405. thanks for your thorough review and catching this Herbert.

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml:145]: 405 is not a valid response code. have removed it from the yaml and also cleaned up the unused 409 code definition.

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml]: commit number 8f08f70

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml]: @hdamker have pushed fix for this for your review.

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml]: totally understand. I myself am having a hard time keeping a track of all the updates against each of the review comments. We do the best we can but Its natural to find bugs at a later point in time and we will address them.

while working on the review comments i took an opportunity to think through what i would do as a developer and error response 400 indicating its a bad request felt like a good fit @hdamker .
below example in Generic400 seemed like a good fit for this scenario.
GENERIC_400_INVALID_ARGUMENT:
              description: Invalid Argument. Generic Syntax Exception

I like keeping the response body of GET and request body of PATCH consistent. In my opinion it makes a developer's life easy. In a scenario where there is a new version of the application where the latency threshold has been more relaxed, the developer would simply do a GET, use the response body to modify the minimum latency value and push that as the request body in a PATCH call without having to recreate the object.

I will consider updating to createApplicationProfile as part of another review comment(#79 ) recommending to create a generic applicationProfile object.

Comment (hdamker) [reply code/API_definitions/application-profiles.yaml]: Feel free to resolve the conversation.

Comment (maheshc01) [reply code/API_definitions/application-profiles.yaml]: thank you @hdamker . Marking the conversation resolved.

**Comment by maheshc01** [comment]
@hdamker with the latest commit. all the issues identified have been fixed with respective commits.
I understand it will be a lot to review each commit but could go over the spec at a high level and confirm if everything looks good?

@Kevsy for additional review and feedback.

File: code/API_definitions/connectivity-insights.yaml
Comment (hdamker) [inline code/API_definitions/connectivity-insights.yaml:206]: I've seen the line because of the word "policy". But on a second look: shouldn't the time stamp be part of the response, not the request?

```suggestion
            current date and time is used and network data for the monitoring
            data aggregation is used to check network performance against the
            application profile defined.
```

**Comment by hdamker** [comment]
> I understand it will be a lot to review each commit but could go over the spec at a high level and confirm if everything looks good?

@maheshc01 yes, I can confirm on a high level view that the spec now looks good to me - but thanks for the understanding that I can't check the details and consistency. 

Looking on the points relevant for Release Management I would ask you to reconsider the content of the CHANGELOG.md. As this is the first release of all three APIs - at least in CAMARA - and the first entry within the CHANGELOG.md some of the listed changes aren't relevant for the reader. E.g. I would change the lines 

> - This version is primarily to align with the changes to Commonalities 0.4-rc.1 and Identity and Consent Management v0.2.0-rc.1.
> - This is the first release of the API.

Into something like:

`This is the first release of the API. The version is aligned with Commonalities 0.4.0-rc.2 and Identity and Consent Management v0.2.0-rc.2.`

(the second sentence is even not needed as it is already noted on the top of the release note)

and maybe list under "Added" the main functionality of the API (or even nothing ... as all is new).

File: code/API_definitions/connectivity-insights.yaml
Comment (maheshc01) [reply code/API_definitions/connectivity-insights.yaml:206]: the complete description for the parameter is provided below:
"this is an optional input parameter. A future data and time can be provided for predictive data. If no value is provided then the current date and time is used and network data for the monitoring data aggregation is used to check network performance against the policy defined."

As per the description this parameter is optional field and can be used to provide a future data and time for querying the predictive KPIs from the network.  Hence the timestamp input is valid as part of the request body.

i will go ahead and update the term policy to application profile as rightly recommended in the review. But please note that there are other places in the yaml where the term policy is used and that reflects the individual network KPI thresholds requested by app developer. i will put in a new enhancement issue created to see if we can revisit the term and see if we can find a better developer friendly word.

**Comment by maheshc01** [comment]
> > I understand it will be a lot to review each commit but could go over the spec at a high level and confirm if everything looks good?
> 
> @maheshc01 yes, I can confirm on a high level view that the spec now looks good to me - but thanks for the understanding that I can't check the details and consistency.
> 
> Looking on the points relevant for Release Management I would ask you to reconsider the content of the CHANGELOG.md. As this is the first release of all three APIs - at least in CAMARA - and the first entry within the CHANGELOG.md, some of the listed changes aren't relevant for the reader. E.g. I would change the lines
> 
> > * This version is primarily to align with the changes to Commonalities 0.4-rc.1 and Identity and Consent Management v0.2.0-rc.1.
> > * This is the first release of the API.
> 
> Into something like:
> 
> `This is the first release of the API. The version is aligned with Commonalities 0.4.0-rc.2 and Identity and Consent Management v0.2.0-rc.2.`
> 
> (the second sentence is even not needed as it is already noted on the top of the release note)
> 
> and maybe list under "Added" the main functionality of the API (or even nothing ... as all is new).

@hdamker 
I will go ahead with the recommended the changes as per this comment- "This version is primarily to align with the changes to Commonalities 0.4-rc.1 and Identity and Consent Management v0.2.0-rc.1."

I need further clarification on content for "Added" and "Changed" sections. The current content in the 2 sections reflect the scope identified for the FALL meta release which is again part of the release management process. 
Hence even though this is the first version of the API, there was a scope identified based on the initial contribution and what is required by the commonalities group.
From that perspective that content in Added and Changed is still valid. 
I am good with either way, but with this additional context could you share your recommendation on what the change log should contain.

**Comment by hdamker** [comment]
@maheshc01 I have also no strong opinion about the content of the CHANGELOG for the first release (and it will be same for the public release which will follow already next week). There is just no release to which you compare, so everything is new, nothing changed, nothing added.

Regarding the approval from Release Management: we discussed today within the Release Management call that @rartych will take over the last review run across the APIs, focussed on Release Management topics, so he is now assigned to https://github.com/camaraproject/ReleaseManagement/issues/74. It would be great if we can get this done within tomorrow's business day, including the Codeowner approval by @Kevsy, the merge and the creation of the release, so that we have a clear state for the TSC on Thursday.

**Comment by maheshc01** [comment]
> @maheshc01 I have also no strong opinion about the content of the CHANGELOG for the first release (and it will be same for the public release which will follow already next week). There is just no release to which you compare, so everything is new, nothing changed, nothing added.
> 
> Regarding the approval from Release Management: we discussed today within the Release Management call that @rartych will take over the last review run across the APIs, focussed on Release Management topics, so he is now assigned to [camaraproject/ReleaseManagement#74](https://github.com/camaraproject/ReleaseManagement/issues/74). It would be great if we can get this done within tomorrow's business day, including the Codeowner approval by @Kevsy, the merge and the creation of the release, so that we have a clear state for the TSC on Thursday.

Thanks for the feedback @hdamker.
in that case i will leave the content of Added and Changed as is in the changelog as it reflects the changes done as defined in the scope of the fall meta release.

@Kevsy could you approve and merge the change and help me with creating the release.

**Comment by hdamker** [comment]
@Kevsy Just wait shortly for the final review by @rartych - he will do the approval on behalf of @camaraproject/release-management_maintainers as @tanjadegroot is still on vacation.

**Comment by rartych** [comment]
@Kevsy Small typo correction: in application-profiles-API-Readiness-Checklist.md line 3: 

Checklist for application-profile**s** v0.3.0-rc.1

API name is application-profiles - plural

**Comment by rartych** [comment]
The  _# Authorization and Authentication_ section in `info.description` needs to be updated to the latest version of the template defined in ICM  in all .yaml files - see the explanation in https://github.com/camaraproject/IdentityAndConsentManagement/issues/185

The template: https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#mandatory-template-for-infodescription-in-camara-api-specs

File: code/API_definitions/application-profiles.yaml
Comment (rartych) [inline code/API_definitions/application-profiles.yaml:62]: As `api-name` is _application-profiles_
```suggestion
            - application-profiles:write
```

File: code/API_definitions/application-profiles.yaml
Comment (rartych) [inline code/API_definitions/application-profiles.yaml:103]: As `api-name` is _application-profiles_

```suggestion
            - application-profiles:write
```

File: code/API_definitions/application-profiles.yaml
Comment (rartych) [inline code/API_definitions/application-profiles.yaml:191]: As `api-name` is application-profiles
```suggestion
            - application-profiles:delete
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:114]:
```suggestion
            - connectivity-insights-subscriptions:create
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:955]: https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#122-event-notification
 For consistency across API we mandate following pattern: `org.camaraproject.<api-name>.<api-version>.<event-name>`
The **type** field is prefixed with a reverse-DNS name, so it is case insensitive, but convention is to use lower case 


```suggestion
      enum:
        - org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met
        - org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met
        - org.camaraproject.connectivity-insights-subscriptions.v0.subscription-ends
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:217]:
```suggestion
            - connectivity-insights-subscriptions:read
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:251]:
```suggestion
            - connectivity-insights-subscriptions:read
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:286]:
```suggestion
            - connectivity-insights-subscriptions:delete
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:793]:
```suggestion
            example: org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:946]: This was changed in Commonalities 0.4.0-rc.2

```suggestion
    EventTypeNotification:
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:887]: It looks that each event type has separate component defined:
```suggestion
          org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met:
            "#/components/schemas/EventNetworkQualityMet"
           org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met:
            "#/components/schemas/EventNetworkQualityNotMet"
```
Following suggestions for lines 891-898

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:898]: New components to support discriminator:

```suggestion
  EventNetworkQualityMet:
      description: event structure for Network Quality met
      allOf:
        - $ref: "#/components/schemas/CloudEvent"
        - type: object
          properties:
            data:
              $ref: "#/components/schemas/NetworkQualityInsight"
              
  EventNetworkQualityNotMet:
      description: event structure for Network Quality not met
      allOf:
        - $ref: "#/components/schemas/CloudEvent"
        - type: object
          properties:
            data:
              $ref: "#/components/schemas/NetworkQualityInsight"              
              
```

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (rartych) [inline code/API_definitions/connectivity-insights-subscriptions.yaml:114]: https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#apis-which-deal-with-explicit-subscriptions

```suggestion
            - connectivity-insights-subscriptions:org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met:create
            - connectivity-insights-subscriptions:org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met:create

```

File: CHANGELOG.md
Comment (rartych) [inline CHANGELOG.md:35]: "Commonalities 0.4.0-rc.2 and Identity and Consent Management v0.2.0-rc.2" should be "upgraded" in lines 25-26 (29-30)

Comment (maheshc01) [reply CHANGELOG.md:35]: issue #94 created to track this.

Comment (maheshc01) [reply CHANGELOG.md:35]: fix available in commit [630372b](https://github.com/camaraproject/ConnectivityInsights/pull/67/commits/630372b35ec23abd958e129a5c45e49e1876354a)

File: code/API_definitions/connectivity-insights-subscriptions.yaml
Comment (maheshc01) [reply code/API_definitions/connectivity-insights-subscriptions.yaml]: @rartych please review this accept of the change in the latest commit. i have reduced it back to 2 events only. first to notify on the network quality and other for subscription end. 
In the same EventNetworkQuality the component support enum of either meets the requirement or does not meet the requirement for each network KPI defined. So 2 different components are not required. 

I will create a separate issue to track this, rather than marking this unresolved. 
issue #95 created for tracking this.

**Comment by maheshc01** [comment]
hi @rartych ,
thank you for the review comments. Have accepted all of your commit suggestions and commits additional changes against issues created based on your review comments. 
Please take a closer look at issue #95 . Need your inputs on that in case i missed something.

Thank you!

**Review by rartych — APPROVED** [review]
LGTM

**Comment by maheshc01** [comment]
Hi @Kevsy i see @rartych  has approved the PR. Could you approve the changes, merge and advice on the next step of release mgmt.

**Review by maheshc01 — APPROVED** [review]

**Comment by hdamker** [comment]
@Kevsy will you also create the r1.1 release or should I take that for you?

**Comment by Kevsy** [comment]
@herbert @mahesh I  can do it now :)

**Comment by Kevsy** [comment]
@hdamker r1.1 released :white_check_mark:

@mahesh you can now proceed with the PR for M4

**Comment by hdamker** [comment]
@maheshc01 @Kevsy Would be good if you could correct the following within the CHANGELOG for r1.1  in course of the M4 release PR (versions should all have -rc.1 as appendix). There is always something which makes it through the reviews :-) 

> r1.1 - rc
> Release Notes
> 
> This release contains the definition and documentation of
> Connectivity Insights API v0.4.0
> Connectivity Insights Subscriptions API v0.4.0
> Application Profiles API v0.3.0
> 

Within the GitHub release description I have already edited it.
Release tracker pages updated with the release tag r1.1

**Pull request merged**
